File: src/test/scala/simulations/RegisterUserSimulation_.scala
//package simulations
//
//import io.gatling.core.Predef.*
//import io.gatling.core.structure.PopulationBuilder
//import io.gatling.http.Predef.*
//import io.gatling.http.protocol.HttpProtocolBuilder
//import scenarios.RegisterUser
//import utils.SetUp_.{rampDurationInMin, rampUp, users}
//import utils.{Headers, SetUp_}
//
//import scala.concurrent.duration.*
//import scala.language.postfixOps
//
//
//class RegisterUserSimulation_ extends Simulation {
//
//  val httpConfiguration: HttpProtocolBuilder = http.baseUrl(SetUp_.baseURL).headers(Headers.requestHeaders)
//    .disableCaching
//    .disableWarmUp
//    .silentResources
//    .perUserNameResolution
//
//  val RegisterUsers: PopulationBuilder =
//        RegisterUser.registerUser().injectOpen(rampUsers(users) during (rampUp minutes))
//          .throttle(reachRps(1) in (60 seconds), holdFor(rampDurationInMin minutes))
//  setUp(RegisterUsers)
//    .protocols(httpConfiguration)
//    .assertions(global.failedRequests.count.is(0))
//}

File: src/test/scala/simulations/RegisterUserSimulation.java
package simulations;


import static io.gatling.javaapi.core.CoreDsl.*;
import static io.gatling.javaapi.http.HttpDsl.*;
import static utils.Header.getAcceptHeaders;

import io.gatling.javaapi.core.*;
import io.gatling.javaapi.http.*;

import scenarios.RegisterUser;

public class RegisterUserSimulation extends Simulation {

    private final String baseURL = utils.SetUp.baseURL;
    private final int users = utils.SetUp.users;
    private final int rampUp = utils.SetUp.rampUp; 
    private final int rampDurationInMin = utils.SetUp.rampDurationInMin; 
    HttpProtocolBuilder httpConfiguration = http.baseUrl(baseURL)
            .headers(getAcceptHeaders()) 
            .disableCaching()
            .disableWarmUp()
            .silentResources()
            .perUserNameResolution();
    
    PopulationBuilder registerUsers = RegisterUser.registerUser()
            .injectOpen(rampUsers(users).during(rampUp)).throttle(reachRps(1).in(60), 
                    holdFor(rampDurationInMin));
    
    {
        setUp(registerUsers)
                .protocols(httpConfiguration)
                .assertions(global().failedRequests().count().is(0L));
    }
}